### DISCUSSION TITLE
Select options for varchar field

### DISCUSSION THREAD

**Discussion by khaliullin** [aminalaee/sqladmin#601] [Q&A] [answered]
Could not find any examples of making select input with options instead of text input for varchar columns.

I have enum:
```
class TaskDifficultyType(str, enum.Enum):
    LIGHT = 'LIGHT'
    DIFFICULT = 'DIFFICULT'
```
and model (I am using sqlmodel):
```
class TaskDifficulty(SQLModel, table=True):
    title: str = Field(nullable=False)
    color: str = Field(nullable=False)
    type: TaskDifficultyType = Field(nullable=True)
```

I'd like to add select input for type field, in Django I would do this:
```
type = models.CharField(
    'Difficulty type',
    choices=TaskDifficultyType.as_choices(),
    max_length=255,
    null=True,
)
```

Is it possible to implement with sqlmodel?

**Comment by aminalaee** [comment, marked as answer]
I don’t really know about SQLModel but if you define your column as enums it will be displayed as a select option.

**Reply by khaliullin** [reply]
Thank you! Also this answer was very useful
https://github.com/tiangolo/sqlmodel/issues/96#issuecomment-1129203076
